core/src/main/java/com/sequenceiq/cloudbreak/service/customconfigs/CustomConfigurationsService.java

```diff
@@ -93,16 +129,17 @@ public CustomConfigurations create(CustomConfigurations customConfigurations, St
                         + " exists. Provide a different name"); });
         initializeCrnForCustomConfigs(customConfigurations, accountId);
         customConfigurations.setAccount(accountId);
-        customConfigurations.getConfigurations().forEach(config -> config.setCustomConfigs(customConfigurations));
+        Set<CustomConfigurationProperty> configurationProperties = new HashSet<>(customConfigurations.getConfigurations());
+        configurationProperties.forEach(config -> config.setCustomConfigs(customConfigurations));
         try {
             transactionService.required(() -> {
-                CustomConfigurations created = customConfigurationsRepository.save(customConfigurations);
-                ownerAssignmentService.assignResourceOwnerRoleIfEntitled(ThreadBasedUserCrnProvider.getUserCrn(), created.getCrn(),
+                customConfigurationPropertyRepository.saveAll(configurationProperties);
+                ownerAssignmentService.assignResourceOwnerRoleIfEntitled(ThreadBasedUserCrnProvider.getUserCrn(), customConfigurations.getCrn(),
```

Comment:
I haven't noticed this until now, but we don't really do transactions where an outside service call is involved. The reason behind this is that it can take a long time, while there might be some lock introduced to the database. This could cause eventually deadlock or performance degradation.
My first idea is to do the db update if the service call is successful, so you won't need any transaction here. If you must do the db change before, you could write the rollback manually if the service call fails.

Reply:
Updated and removed the transaction, as discussed.